includes/admin/payments/view-order-details.php

```diff
@@ -512,7 +512,17 @@
 										<div class="column">
 											<p>
 												<strong><?php esc_html_e( 'Donor Name:', 'give' ); ?></strong><br>
-												<?php echo $customer->name; ?>
+												<?php
+                                                $donor_name = give_get_donor_name_by( $payment_id, 'donation' );
+                                                $customer_name = give_get_donor_name_by( $customer_id, 'donor' );
+
+                                                // Check whether the donor name and WP_User name is same or not.
+                                                if( sanitize_title( $donor_name ) != sanitize_title( $customer_name ) ){
```

Comment:
@ravinderk Updated code with outputing donor billing name + donor name on detail page as well and similar to listing page